tools/toolchains/iar.py

```diff
@@ -47,10 +47,12 @@ def __init__(self, target, notify=None, macros=None, build_profile=None,
                                build_profile=build_profile)
         if target.core == "Cortex-M7F" or target.core == "Cortex-M7FD":
             cpuchoice = "Cortex-M7"
-        elif target.core.startswith("Cortex-M23"):
-            cpuchoice = "8-M.baseline"
+        elif target.core.startswith("Cortex-M33FD"):
```

Comment:
It's not being introduced by this PR, but I'm not a huge fan of this "D" meaning something different to the "D" in "M7FD".

In the past ARM used "E" to indicate the DSP extension, so M33FE might make more sense?

Although I'd be inclined to start following the modern tools' lead and go with "Cortex-M33.no_dsp.no_fp" - pick either the armcc or gcc form to use for Mbed OS, rather than making up our own single-letter suffixes.

Reply:
@kjbracey-arm - Agree to you on this, but I will leave this out of the scope of this PR.

@SenRamakri - It will be good to add this as task in tools. It will impact all M devices with different variants and all toolchians, part of tools cleanup